torchtune/utils/_device.py

```diff
@@ -156,3 +175,54 @@ def batch_to_device(batch: dict, device: torch.device) -> None:
                 f"""To use batch_to_device, all elements in the batch must be a dict or Tensor.
 Got key "{k}" with value of type {type(v)}"""
             )
+
+
+class DeviceSupport(Enum):
+    """
+    This is a simple enum for compute devices,
+    This currently only supports CPU, CUDA, NPU.
+    """
+
+    CPU = ("cpu", "CPU", "gloo")
+    CUDA = ("cuda", "GPU", "nccl")
+    NPU = ("npu", "NPU", "hccl")
```

Comment:
Do we also need an item for MPS here? cc @SalmanMohammadi

Reply:
Also more of a nit, but I wonder if we should use a NamedTuple abstraction here. (Alternatively can just add a comment explaining what each of the fields correspond to)

Reply:
Thank you so much for your suggestions! I’ll make sure to add the relevant comments here.